src/parse.coffee

```diff
@@ -14,9 +13,11 @@ createWarning = (message) ->
 
 parse = (source, callback) ->
   warning = null
-  mediaType = deckardcain.identify(source)
+  adapters = fury.detect(source)
 
-  unless mediaType
+  if adapters.length
+    mediaType = adapters[0].mediaTypes[0]
```

Comment:
Just a heads up, mediaTypes is array so it could contain multiple things:

- `application/swagger+json ` and `application/swagger+yaml` https://github.com/apiaryio/fury-adapter-swagger/blob/master/src/adapter.js#L8
- `text/vnd.apiblueprint` `text/vnd.apiblueprint+markdown` https://github.com/apiaryio/fury-adapter-apib-parser/blob/master/src/adapter.js#L8

Seems it works fine in your case as first is the one you check but, just to note that it might be fragile, if you want to match against specific adapter you are best comparing the adapter == the adapter. F.e:

```js
adapter == require('fury-adapter-apib-parser');
```